### ISSUE TITLE
Getting Rancher working with Native Docker for Desktop Kubernetes

### ISSUE THREAD

**Issue by scriptjs** [rancher/rancher#14358]
Hello, I've been working to understand how we can get Docker for Desktop working with the local and native Kubernetes cluster shipped with Docker Edge. The issue with Docker for Desktop is its networking. There is no docker0 interface bridged to the host. Kubernetes on Docker for Desktop uses a default node IP of 192.168.65.3 assigned from the Docker subnet of 192.168.65.0 for Docker for Desktop.

Docker for Desktop utilizes some magic to expose docker containers for the host so it is accessible on the host ip, localhost, and loopback 127.0.01 interface. For example, a workload that exposes port 30000 is accessible at 127.0.0.1:30000. This doesn't work for Docker for Desktop and Rancher since the public address is seen as 192.168.65.3 without a bridge. This impacts the urls and configuration for HostIP, NodeIP, ClusterIP and L7 load balancing to expose the workloads in Rancher.

While experimenting with the issue with a workload, I thought of aliasing the local interface of the machine with the node IP. Constructing links, I could get the workload to display properly. I have tested on MacOS creating the alias as follows:

```
sudo ifconfig lo0 alias 192.168.65.3
```
Windows can use an alternate command such as `netsh` to alias the local interface in an admin escalated terminal. 

Evaluating the YAML, I have compared creation of a workload using HostPort, NodePort and ClusterIP to see what happens to the workload YAML after a L7 load balancer has been configured for the workload.

The issue is that the publicEndpoints metadata is not written and the public address is omitted in the configuration.

To illustrate creating a workload using ClusterIP, there is no issues with the initial creation because ClusterIP initially provides no links in the UI for the workload . No publicEndpoints metadata is added until an ingress is exposed through a L7 load balancer.

Normally, when a load balancer is created, a metadata annotation is created in its configuration YAML for field.cattle.io/publicEndpoints. Because it is unable to get the publicIP, it omits this necessary metadata. Second, it would normally update the workload configuration on load balancer creation with a list of field.cattle.io/publicEndpoints. Because it cannot get access to the public address, it fails to include the addresses property for each internal port. Also, because it cannot set the field.cattle.io/publicEndpoints on the load balancer, it omits these items in the list for public ports in the field.cattle.io/publicEndpoints on the workload.

**Solution**
The solution the problem for the Docker for Desktop is to have an import property that could be checked on the UI or set in Rancher. Setting this would perform a test of the node IP for Docker for Desktop Kubernetes and the IP of the local interface. Rancher would retain these IPs as metadata. In next step, in the UI, in addition to the kubectl command to import the cluster, Rancher could display the command to set the alias for the local interface with the Docker for Desktop Kubernetes node ip.

The user imports the Docker for Desktop Kubernetes with the kubectl command and executes the alias command.

As far as changes to the code, where public addresses are set in Rancher server logic, the Docker for Desktop IP metadata would be used to substitute the node IP with the local interface IP. This will correct the configuration and urls, and it will work.

**Other notes**
What I can say is that creating and destroying resources using Rancher on Docker for Desktop Kubernetes with 4GB RAM is quite a bit more performant than using MiniKube or an Ubuntu VM creating a single local cluster in Fusion or VirtualBox. So, getting this fixed will mean an easy and lightweight path to working with Rancher Server natively with a local and native Kubernetes.

**Rancher versions:**
rancher/server or rancher/rancher: 2.04
rancher/agent or rancher/rancher-agent: 2.04

**Infrastructure Stack versions:**
healthcheck:
ipsec:
network-services:
scheduler:
kubernetes (if applicable): v1.10.3

**Docker version: (`docker version`,`docker info` preferred)** 18.05.0-ce

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
MacOS High Sierra

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node

**Comment by xiaods** [comment]
yes, this is cool feature for testing or devel environment for mac.

**Comment by xiaods** [comment]
https://rancher.com/blog/2018/2018-05-18-how-to-run-rancher-2-0-on-your-desktop/  this is match your requestments?

**Comment by xiaods** [comment]
any update on this topic?

**Comment by brandon-carag** [comment]
Any update on this?

**Comment by fzan** [comment]
updates?

**Comment by loganhz** [comment]
> yes, this is cool feature for testing or devel environment for mac.

You can use https://github.com/rancher/k3d to create a k3s cluster and import into rancher for dev env
